Mathlib/RingTheory/RootsOfUnity/PrimitiveRoots.lean

```diff
@@ -286,6 +286,29 @@ theorem map_iff_of_injective [MonoidHomClass F M N] (hf : Injective f) :
     IsPrimitiveRoot (f ζ) k ↔ IsPrimitiveRoot ζ k :=
   ⟨fun h => h.of_map_of_injective hf, fun h => h.map_of_injective hf⟩
 
+lemma push {n : ℕ} [NeZero n] {m : M} (hm : IsPrimitiveRoot m n)
```

Comment:
This seems to be a special case of [`IsPrimitiveRoot.map_of_injective`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/RootsOfUnity/PrimitiveRoots.html#IsPrimitiveRoot.map_of_injective)

Reply:
Yes, I've replaced it with `IsPrimitiveRoot.map_of_injective`, thanks.

The remaining proof of `HasEnoughRootsOfUnity.prim` is still rather convoluted - I have difficulty relating equations in `hm.toRootsOfUnity` to equations in `m`. I expect I am missing something here?

Reply:
Thanks for the `HasEnoughRootsOfUnity.prim` golf!